students/prgrover/lesson07/html_render.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python3
+
+class Element:
+    """
+    Base class for rendering an HTML element
+    """
+
+    tag = 'html'
+    indent = '    ' #4 space characters
+
+    def __init__(self, content=None, **kwargs):
+        self.attributes = kwargs
+        self.content = []
+        if content:
+            self.append(content)
+
+
+    def append(self, content):
+        if hasattr(content, 'render'):
+            self.content.append(content)
+        else:
+            self.content.append(TextWrapper(str(content)))
+
+
+    #Per Step 4 - next four methods are an attempt to refactor common parts into separate methods
+    def render(self, file_out, cur_ind=""):
+        self.make_open_tag(file_out, cur_ind)
+        for markup in self.content:
+            markup.render(file_out, cur_ind + self.indent)
+            file_out.write('\n')
+        close_tag = '{}</{}>'.format(cur_ind, self.tag)
+        file_out.write(close_tag)
+
+
+    def make_open_tag(self, file_out, cur_ind="", one_line=False):
```

Comment:
Generally, it's better to override the method in a subclass than have boolean flag for subclass behavior in base class method